File: Data/DataModels/OfferRide.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CarPooling.Data.DataModels
{
    [Table("OfferRides")]
    public class OfferRide
    {
        [Key]
        [Column("OfferRideId")]
        public Guid OfferRideId { get; set; }
        [Column("UserId")]
        public int UserId { get; set; }
        [Column("UserName")]
        public string? UserName { get; set; }
        [Column("StartPoint")]
        [Required]
        public string StartPoint { get; set; } = null!;
        [Column("Stops")]
        public string? Stops { get; set; }
        [Column("EndPoint")]
        [Required]
        public string EndPoint { get; set; } = null!;
        [Column("Date")]
        [Required]
        public string Date { get; set; } = null!;
        [Column("TimeSlot")]
        [Required]
        public string TimeSlot { get; set; } = null!;
        [Column("Seats")]
        [Required]
        public int Seats { get; set; } = 0;
        [Column("Price")]
        [Required]
        public string Price { get; set; } = null!;
    }
}

File: src/app/shared/models/offer-ride.ts
export class OfferRide {
    startPoint: string;
    endPoint: string;
    date: string;
    timeSlot: string;
    seats: number;
    price: string;
    stops: string;
    userName: string;

    constructor(startPoint: string, endPoint: string, date: string, timeSlot: string, seats: number, stops: string, price: string, userName: string
    ) {
        this.startPoint = startPoint;
        this.endPoint = endPoint;
        this.date = date;
        this.timeSlot = timeSlot;
        this.seats = seats;
        this.price = price;
        this.userName = userName;
        this.stops = stops;
    }
}